File: sdk/php/src/Resources/Balance.php
<?php

namespace Pexipay\Resources;

use Pexipay\PexipayClient;

class Balance
{
    private PexipayClient $client;

    public function __construct(PexipayClient $client)
    {
        $this->client = $client;
    }

    public function retrieve(): array
    {
        $response = $this->client->request('GET', '/balance');
        return $response['data'] ?? $response;
    }

    public function listTransactions(array $params = []): array
    {
        return $this->client->request('GET', '/balance/transactions', [
            'query' => $params
        ]);
    }
}

File: docs/sdk/python/pexipay/resources/balance.py
"""Balance resource"""

from typing import TYPE_CHECKING, Optional, Dict, Any

if TYPE_CHECKING:
    from ..client import PexipayClient


class BalanceResource:
    """Balance API resource"""

    def __init__(self, client: "PexipayClient"):
        self.client = client

    def retrieve(self) -> Dict[str, Any]:
        """Retrieve account balance"""
        response = self.client.request("GET", "/balance")
        return response.get("data", response)

    def list_transactions(
        self,
        limit: Optional[int] = None,
        starting_after: Optional[str] = None,
        ending_before: Optional[str] = None,
    ) -> Dict[str, Any]:
        """List balance transactions"""
        params = {
            "limit": limit,
            "startingAfter": starting_after,
            "endingBefore": ending_before,
        }
        # Remove None values
        params = {k: v for k, v in params.items() if v is not None}

        return self.client.request("GET", "/balance/transactions", params=params)
